src/leaflike/user/core.clj

```diff
@@ -0,0 +1,35 @@
+(ns leaflike.user.core
+  (:require [leaflike.user.db :as user-db]
+            [leaflike.user.auth :as auth]
+            [leaflike.user.validator :as validator]))
+
+(defn signup
+  [request]
+
+  (let [body  (clojure.walk/keywordize-keys (-> request :params))
+        valid (validator/valid-registration? body)]
+    (cond
```

Comment:
A cond doesn't give is much here over an if. It perhaps makes the code more verbose here?